Upgrade to Symfony 6

PGBastien opened the pull request:
This PR allow to upgrade Symfony from `5.4` to `6.4`:

- [X] [Upgrade PHP dependencies](https://github.com/najmi9/mercure-symfony-chat/commit/ebbdf02e348e0d1acd5e6fdffb88fc869b673d41)
- [X]  [Upgrade PHP 7.4 to 8.3](https://github.com/najmi9/mercure-symfony-chat/commit/b3274c0cf9ddd5538ad4786fffcd6845585c5b67)
- [X] [Upgrade Symfony 5.4 to 6.4](https://github.com/najmi9/mercure-symfony-chat/commit/4673c722922e2291014d48bce53ec572a129f946)
- [X] Replace PHP annotations by attributes

And a fix on param page undefined.

Actual Symfony LTS = `6.4.2` => https://symfony.com/releases


Source:
- https://symfony.com/doc/6.4/routing.html#creating-routes-as-attributes
- https://symfony.com/doc/6.4/security.html#loading-the-user-the-user-provider
- https://symfony.com/doc/6.4/security/access_control.html
- https://symfony.com/doc/6.4/security/access_denied_handler.html#customizing-all-access-denied-responses
- https://symfony.com/blog/new-in-symfony-5-3-passwordhasher-component
- https://symfony.com/doc/6.4/reference/attributes.html

PGBastien committed:
fix: param page undefined

PGBastien committed:
chore: Upgrade PHP dependencies

PGBastien committed:
chore: Upgrade PHP `7.4` to `8.3`

PGBastien committed:
chore: Upgrade Symfony `5.4` to `6.4`

najmi9 commented on src/Entity/User.php:
```diff
@@ -18,7 +18,7 @@
  * @ORM\Entity(repositoryClass=UserRepository::class)
  * @UniqueEntity(fields={"email"}, message="There is already an account with this email")
```
@PGBastien Can we use php 8 attributes here also instead of annotations??

PGBastien replied:
Hi @najmi9 good idea 👍🏼 
[`7a0b6fa` (#2)](https://github.com/najmi9/mercure-symfony-chat/pull/2/commits/7a0b6faf6f8b75a7ec0ed51c8bd535d387eb7e19)

najmi9 replied:
Thanks 🧡.

PGBastien committed:
feat: From PHP annotations by attributes

This commit revise doctrine mapping type from `annotation` -> `attribute`
and replace annotations by attributes into entities.

Source:
https://symfony.com/doc/6.4/reference/attributes.html

najmi9 reviewed (approved):
Really great PR, thanks.

PGBastien commented:
> Really great PR, thanks.

With pleasure 👍🏼
